kong/resty/dns/client.lua

```diff
@@ -1158,10 +1139,8 @@ local function resolve(qname, r_opts, dnsCacheOnly, try_list, force_no_sync)
     for k,v in pairs(r_opts) do opts[k] = v end  -- copy the options table
   end
 
-  -- default the ADDITIONAL SECTION to TRUE
-  if opts.additional_section == nil then
-    opts.additional_section = true
-  end
+  -- avoid parsing ip addresses that do not match the qname
+  opts.additional_section = nil
```

Comment:
I think that it is the only difference with #13002, could I get some explain?

Reply:
updated in description,

> The regression was that after we merged https://github.com/Kong/kong/pull/13002, the kong DNS client will incorrectly think all the RRs in additional section as results of queried name. This bug is caused by: 
> 1. The DNS client incorrectly using the content of additional section from the response. This was introduced in 3.0 as part of an [perf optimization](https://github.com/Kong/kong/pull/8895)
> 2. The DNS fix introduced in kong:[ fix(dns): ignore records with non-matching types when parsing answers](https://github.com/Kong/kong/pull/13002) which supposedly makes us more standard compliant. However, using any A records from additional section will be a severe bug, (using any A records from the answers section is expected and correct).